docs/source/caching/index.rst

```diff
@@ -0,0 +1,115 @@
+
+Caching is also implemented for Data nodes. This is not very useful in practice (yet), but is an easy way to show how the caching mechanism works:
+
+.. ipython::
+    :verbatim:
+
+    In [1]: from __future__ import print_function
+
+    In [2]: from aiida.orm.data.base import Str
+
+    In [3]: n1 = Str('test string')
+
+    In [4]: n1.store()
+    Out[4]: u'test string'
+
+    In [5]: n2 = Str('test string')
+
+    In [6]: n2.store(use_cache=True)
+    Out[6]: u'test string'
+
+    In [7]: print('UUID of n1:', n1.uuid)
+    UUID of n1: 956109e1-4382-4240-a711-2a4f3b522122
+
+    In [8]: print('n2 is cached from:', n2.get_extra('cached_from'))
+    n2 is cached from: 956109e1-4382-4240-a711-2a4f3b522122
+
+As you can see, passing ``use_cache=True`` to the ``store`` method enables using the cache. The fact that ``n2`` was created from ``n1`` is stored in the ``cached_from`` extra of ``n2``.
+
+When running a ``JobCalculation`` through the ``Process`` interface, you cannot directly set the ``use_cache`` flag when the calculation node is stored internally. Instead, you can pass the ``_use_cache`` flag to the ``run`` or ``submit`` method.
+
+Caching is **not** implemented for workchains and workfunctions. Unlike calculations, they can not only create new data nodes, but also return exsting ones. When copying a cached workchain, it's not clear which node should be returned without actually running the workchain. This is explained in more detail in the section :ref:`caching_provenance`.
+
+Configuration
+-------------
+
+Of course, using caching would be quite tedious if you had to set ``use_cache`` manually everywhere. To fix this, the default for ``use_cache`` can be set in the ``.aiida/cache_config.yml`` file. You can specify a global default, or enable / disable caching for specific calculation or data classes. An example configuration file might look like this:
+
+.. code:: yaml
+
+    profile-name:
+      default: False
+      enabled:
+        - aiida.orm.calculation.job.simpleplugins.templatereplacer.TemplatereplacerCalculation
+        - aiida.orm.data.base.Str
+      disabled:
+        - aiida.orm.data.base.Float
+
+This means that caching is enabled for ``TemplatereplacerCalculation`` and ``Str``, and disabled for all other classes. In this example, manually disabling ``aiida.orm.data.base.Float`` is actually not needed, since the default value for all classes is already ``False``. Note also that the fully qualified class import name (e.g., ``aiida.orm.data.base.Str``) must be given, not just the class name (``Str``). This is to avoid accidentally matching classes with the same name.
+
```

Comment:
Unless the metaclass of the Node has a custom ``__str__``, yes. A more stable way to get it would be ``Str.__module__ + '.' + Str.__name__``.

I now added a snippet on how to get the full name to the documentation.